documentation/DataModel.md

```diff
@@ -6,3 +6,124 @@ configuration, CouchDB-compatible PouchDB is used for the data storage.
 The two GPII components that read and write data from the data storage are Preferences Server and Authorization Server.
 
 The details of the GPII data model can be found [here](https://wiki.gpii.net/w/Keys,_KeyTokens,_and_Preferences).
+
+## Preference Safes Overview
+
+In this section we discuss the CouchDB documents for all the data associated with a users preferences safe. This
+includes their preference sets, keys, full login credentials, and a minimal amount of metadata for the user,
+such as name and email.  Currently, this consists of three document types: `prefsSafe`, `gpiiKey`,
+`gpiiCloudSafeCredential`.
+
+### Preference Safes
+
+Preference Safes consist of a single primary document of type `prefsSafe`. These contain some optional metadata such
+as `name` and `email`, and the `preferences` section which contains the users preference sets. This is the central
+document for a safe, any documents relating to a safe should have a property `prefsSafeId` which contains the id of
+the preferences safe.
+
+An example document:
+
+```json
+    {
+        "_id": "prefsSafe-7",
+        "type": "prefsSafe",
+        "schemaVersion": "0.1",
+        "prefsSafeType": "user",
+        "name": null,
+        "email": null,
```

Comment:
It just came to me that, with the data model change (the removal of "password" field from "prefsSafe" document type and the addition of "timestampRevoked" into "gpiiKey" type), other relevant places in the repo should be changed accordingly. This includes [the conversion script](https://github.com/GPII/universal/blob/master/scripts/shared/prefsSetsDbUtils.js#L73) that converts preferences json files at testData/preferences to couchDB document structures as well as other data used for unit and integration tests. Doing a search using keywords like "password", "gpiiKey" will reveal more.

Reply:
Good point.  I've removed all the old `password` fields from everywhere: code, scripts, and tests.  I'm not familiar with the timestamp revoked portions of the keys as I don't believe this ticket/work has touched that, so I've left gpiiKey as is.